angr/project.py

```diff
@@ -693,28 +693,16 @@ def __repr__(self):
     # Properties
     #
 
-    @property
-    def use_sim_procedures(self):
-        return self._should_use_sim_procedures
-
-    @property
-    def is_java_project(self):
-        """
-        Indicates if the project's main binary is a Java Archive.
-        """
-        if self._is_java_project is None:
-            self._is_java_project = isinstance(self.arch, ArchSoot)
-        return self._is_java_project
+    def __getattr__(self, a):
```

Comment:
This is magic -- why can't we have them initialized inside `__init__()`?

Reply:
I don't know this code, and it maintains the semantics. I can change it to __init__, but I'm not going to chase down whatever java issues arise. My suggestion is to do this first, and do that change under a different PR.

Reply:
I'm moving it to `__init__()` and as far as I know nothing should break. CI will tell us if I'm right.